pkg/util/k8sutil/k8sutil.go

```diff
@@ -185,15 +219,28 @@ func CreateClientService(ctx context.Context, kubecli kubernetes.Interface, serv
 		} else {
 			clientPorts = defaultPort
 		}
-		err = createService(ctx, kubecli, serviceName, clusterName, ns, clientPorts, owner, false, service)
+		err = createSvc(ctx, kubecli, serviceName, clusterName, ns, clientPorts, owner, false, service, service.Annotations)
 	} else {
-		err = createService(ctx, kubecli, serviceName, clusterName, ns, defaultPort, owner, false, nil)
+		annotations := map[string]string{}
```

Comment:
I usually prefer instead of adding many if else, to extra the code in a function that for example do this:

if cluster mode = disc:
  service = set service
  annotations = set annotations 
  return service, annotations
return nil, nil

Reply:
Don't know if I agree since its just plain value creation using straight typing or struct assignment